### PR TITLE
[api-migration-hackathon] TableDataSelector migration

### PR DESCRIPTION
**Pull request by tbmasood**
Migrated the TableDataSelector module to the new API specification. I have successfully compiled and tested this module as best as I could, but it would be nice if some expert have a look if all the changes have been done properly, because I don't know much about the functionality of this module.

None of state files in the data repo are affected.

### CODE REVIEW

File: core/vtk/ttkTableDataSelector/ttkTableDataSelector.h [hunk 1]
```diff
@@ -99,7 +66,10 @@ class TTKTABLEDATASELECTOR_EXPORT ttkTableDataSelector
 
 protected:
   ttkTableDataSelector() {
-    UseAllCores = true;
+    this->setDebugMsgPrefix("ttkTableDataSelector");
```
Comment (julien-tierny) [inline core/vtk/ttkTableDataSelector/ttkTableDataSelector.h]: same remark as in PR #403 (```TableDataSelector```)

File: core/vtk/ttkTableDataSelector/ttkTableDataSelector.cpp [hunk 2]
```diff
@@ -7,45 +9,58 @@ using namespace ttk;
...
+  return ttkAlgorithm::RequestInformation(request, inputVector, outputVector);
+}
+
+int ttkTableDataSelector::RequestData(vtkInformation *request,
+                                      vtkInformationVector **inputVector,
+                                      vtkInformationVector *outputVector) {
+  vtkTable *input = vtkTable::GetData(inputVector[0]);
+  vtkTable *output = vtkTable::GetData(outputVector);
+
   Memory m;
```
Comment (julien-tierny) [inline core/vtk/ttkTableDataSelector/ttkTableDataSelector.cpp]: same remark as in PR #403 (remove memory trackers)

**Comment by julien-tierny** [comment]
hi talha,

thanks a lot for the PR.

could please address the minor remarks I posted above?
cheers,
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

**Comment by tbmasood** [comment]
Requested changes have been done.

**Comment by julien-tierny** [comment]
awesome, thanks!
julien
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

**Pull request merged**